src/Symfony/Component/Translation/Translator.php

```diff
@@ -72,17 +75,25 @@ class Translator implements TranslatorInterface, TranslatorBagInterface
     /**
      * Constructor.
      *
-     * @param string               $locale   The locale
-     * @param MessageSelector|null $selector The message selector for pluralization
-     * @param string|null          $cacheDir The directory to use for the cache
-     * @param bool                 $debug    Use cache in debug mode ?
+     * @param string                                         $locale    The locale
+     * @param MessageFormatterInterface|MessageSelector|null $formatter The message formatter
+     *                                                                  Passing the MessageSelector as a second parameter is deprecated since version 3.2
+     * @param string|null                                    $cacheDir  The directory to use for the cache
+     * @param bool                                           $debug     Use cache in debug mode ?
      *
      * @throws \InvalidArgumentException If a locale contains invalid characters
      */
-    public function __construct($locale, MessageSelector $selector = null, $cacheDir = null, $debug = false)
+    public function __construct($locale, $formatter = null, $cacheDir = null, $debug = false)
     {
         $this->setLocale($locale);
-        $this->selector = $selector ?: new MessageSelector();
+
+        if ($formatter instanceof MessageSelector) {
+            $this->formatter = new TransMessageFormatter($formatter);
+            @trigger_error('Passing a MessageSelector instance into the '.__METHOD__.' as a second argument is deprecated since version 3.1 and will be removed in 4.0. Inject a MessageFormatterInterface[] instance instead.', E_USER_DEPRECATED);
```

Comment:
What about:
```php
@trigger_error(sprintf('Passing a "%s" instance into the "%s" as a second argument is deprecated since version 3.2 and will be removed in 4.0. Inject a "%s" implementation instead.', MessageSelector::class, __METHOD__, MessageFormatterInterface::class), E_USER_DEPRECATED)
```
?